File: finkzeit/finkzeit/report/customer_credit_ledger/customer_credit_ledger.py
# Copyright (c) 2020, Fink Zeitsysteme/libracore and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from finkzeit.finkzeit.credit_controller import get_credit_account_ledger
from frappe import _

def execute(filters=None):
    columns, data = [], []

    columns = get_columns()
    if filters.customer:
        data = get_values(filters)
    else:
        data = []
    
    return columns, data

def get_columns():
    return [
        {"label": _("Date"), "fieldname": "date", "fieldtype": "Date", "width": 100},
        {"label": _("Amount"), "fieldname": "amount", "fieldtype": "Currency", "width": 100},
        {"label": _("Balance"), "fieldname": "balance", "fieldtype": "Currency", "width": 100}
    ]

def get_values(filters):
    data = get_credit_account_ledger(customer=filters.customer, date=filters.date)
    return data

File: finkzeit/finkzeit/report/customer_credit_ledger/customer_credit_ledger.js
// Copyright (c) 2020, Fink Zeitsysteme/libracore and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Customer Credit Ledger"] = {
	"filters": [
		{
			fieldname:"customer",
			label: __("Customer"),
			fieldtype: "Link",
            options: "Customer",
            reqd: 1
        },
        {
			fieldname:"date",
			label: __("Date"),
			fieldtype: "Date"
        }
	]
};
